crates/symbolicator-service/src/caching/memory.rs

```diff
@@ -209,26 +210,36 @@ impl<T: CacheItemRequest> Cacher<T> {
     ///
     /// This method does not take care of ensuring the computation only happens once even
     /// for concurrent requests, see the public [`Cacher::compute_memoized`] for this.
-    async fn compute(
-        &self,
-        request: T,
-        key: &CacheKey,
-        is_refresh: bool,
-    ) -> CacheContents<T::Item> {
+    async fn compute(&self, request: T, key: &CacheKey, is_refresh: bool) -> CacheEntry<T::Item> {
         let name = self.config.name();
         let cache_path = key.cache_path(T::VERSIONS.current);
-        let mut temp_file = self.config.tempfile()?;
+        let mut metadata = None;
+        let mut temp_file = match self.config.tempfile() {
+            Ok(temp_file) => temp_file,
+            Err(e) => return CacheEntry::from_err(e),
+        };
 
         let shared_cache = self.shared_cache(&request);
         let shared_cache_hit = if let Some(shared_cache) = shared_cache {
-            let temp_fd = tokio::fs::File::from_std(temp_file.reopen()?);
+            // TODO: Here we create fresh metadata upon fetching from the shared cache.
+            // Ideally we would obviously want to share these metadata, but this is left
+            // as future work.
```

Comment:
Good question, how much we would want to share here. File specific metadata of course, but things like the created timestamp we wouldn't want to necessarily use for on disk cache duration (except maybe for a total max ttl).

Which leads me more to think about again that we might want backend specific metadata and shared metadata. If a backend needs additional metadata to work (Filesystem may need access times), they shouldn't bleed into other backends.